packages/components/src/dashicon/index.tsx

```diff
@@ -7,12 +7,23 @@
  */
 /** @typedef {import('react').ComponentPropsWithoutRef<'span'> & OwnProps} Props */
 
+/**
+ * Internal dependencies
+ */
+import type { DashiconProps } from './types';
+
 /**
  * @param {Props} props
  * @return {JSX.Element} Element
  */
 
-function Dashicon( { icon, className, size = 20, style = {}, ...extraProps } ) {
+function Dashicon( {
+	icon,
+	className,
+	size = 20,
+	style = {},
+	...extraProps
+}: DashiconProps ) {
```

Comment:
You don't have to go through the entire [TS migration guide](https://github.com/WordPress/gutenberg/blob/trunk/packages/components/CONTRIBUTING.md#refactoring-a-component-to-typescript) because Dashicon component is really simple and doesn't have any stories/tests. But it might help to see item `vi` of bullet point 7, where it describes how we use the `WordPressComponentProps` utility type for cases like this, instead of `ComponentPropsWithoutRef`.

You can see some usage examples in other components, for example:

https://github.com/WordPress/gutenberg/blob/367c180b384f6656aa6fa5f6926e0c39ab073905/packages/components/src/search-control/index.tsx#L36

So here that would look like:

```suggestion
}: WordPressComponentProps< DashiconProps, 'span', false > ) {
```